src/lib.rs

```diff
@@ -262,19 +291,62 @@ impl Scope {
             .or_insert_with(|| Import::new(path, ty))
     }
 
+    /// Returns true if a module named `name` exists in this scope.
+    pub fn contains_module<Q: ?Sized>(&self, name: &Q) -> bool 
+    where
+        Q: Hash + ordermap::Equivalent<RcKey>
+    {
+        self.modules.contains_key(name)
+    }
+
     /// Push a new module definition, returning a mutable reference to it.
+    /// 
+    /// # Note
+    /// 
+    /// Since a module's name must uniquely identify it within the scope in 
+    /// which it is defined, calling this function with a name already used 
+    /// in this scope will (silently) clobber the existing module definition.
+    /// 
+    /// In many cases, the [`module_or_add`] function is preferrable, as it will
+    /// return the existing definition instead.
+    /// 
+    /// [`module_or_add`]: #method.module_or_add
     pub fn new_module(&mut self, name: &str) -> &mut Module {
         self.push_module(Module::new(name));
 
-        match *self.items.last_mut().unwrap() {
-            Item::Module(ref mut v) => v,
-            _ => unreachable!(),
+        &mut self.modules[name]
+    }
+
+    /// Returns a mutable reference to a module if it is exists in this scope. 
+    pub fn module_mut(&mut self, name: &str) -> Option<&mut Module> {
+        self.modules.get_mut(name)
+    }
+
+    /// Returns a mutable reference to a module, creating it if it does 
+    /// not exist.
+    pub fn module_or_add(&mut self, name: &str) -> &mut Module {
```

Comment:
I'd get rid of this for now. The check can be done w/ other functions.

If this becomes a pattern, then we can look at all the instances (modules, structs, etc...) and see if we can come up with a good name.

Reply:
Hmm...I opened the PR with the primary goal of adding this function specifically, because I needed it in my changes to `tower-grpc` (`get_module` and `get_module_mut` were added only because it was more or less trivial to add them as well while making this change), and currently, it's the only one of these functions that's currently used in a consumer of `codegen` (my branch of `tower-grpc`). 

While it's certainly possible to remove this & rewrite the functions that use it, I think that code is a bit clearer with this function, and my suspicion is that it's it's likely to be the most common use-case for retrieving a module...

Reply:
(now that `module_mut` has been renamed to `get_module_mut` (299c69affe505d0f1e55ea17853cdf7ed919a678), what about renaming this to just `module`, by analogy with `import`, which also has similar "get or else add" behavior?)